packages/discovery-provider/src/gated_content/content_access_checker.py

```diff
@@ -157,65 +161,66 @@ def check_access_for_batch(
         track_access_result: GatedTrackAccessResult = {}
 
         for track_id, track_entity in gated_track_data.items():
-            user_id = track_access_users[track_id]
-            if user_id not in track_access_result:
-                track_access_result[user_id] = {}
-
-            # content owner has access to their own gated content
-            content_owner_id = track_entity["content_owner_id"]
-            if content_owner_id == user_id:
-                track_access_result[user_id][track_id] = {
-                    "has_stream_access": True,
-                    "has_download_access": True,
-                }
-                continue
-
-            # if not gated on either stream or download,
-            # then check if track is a stem track and check parent track access,
-            # otherwise, user has access to stream and download.
-            # note that stem tracks do not have stream/download conditions.
-            stream_conditions = track_entity["stream_conditions"]
-            download_conditions = track_entity["download_conditions"]
-            if not stream_conditions and not download_conditions:
-                access = self._check_stem_access(
-                    session=session,
-                    user_id=user_id,
-                    content_entity=track_entity,
-                )
-                track_access_result[user_id][track_id] = {
-                    "has_stream_access": access,
-                    "has_download_access": access,
-                }
-                continue
-
-            # if stream gated, check stream access which also determines download access
-            if stream_conditions:
-                has_access = self._evaluate_conditions(
+            user_ids = track_access_users[track_id]
```

Comment:
this reads fine to me. but i wonder if we should simplify/clarify/separate track v album logic further, as this is starting to get complex. for example, `check_stem_access` is irrelevant if content type is album